db_classes1.py

```diff
@@ -127,19 +132,24 @@ def get(self, *args, **kwargs):
 
 
 class HistoryVacation:
-    def __init__(self, id, start_date, finish_date, employee_id, database):
+    def __init__(self, id, start_date, finish_date, employee_id, department_id, database):
         self.id = id
         self.start_date = start_date
         self.finish_date = finish_date
         self.employee_id = employee_id
         self.database = database
+        self.department_id = department_id
         self.sql_script = ''
 
     def save(self):
         self.sql_script = f'''INSERT INTO DEPARTMENT (ID, start_date, finish_date, employee_id) 
             VALUES ('{self.id}', '{self.start_date}', '{self.finish_date}', '{self.employee_id}')'''
-        employees_on_vacation = f'''?????????????'''
-        if employees_on_vacation >= 5:
+        employees_on_vacation = f'''SELECT count(HISTORY_VACATION.employee_id) FROM HISTORY_VACATION
+                                    JOIN EMPLOYEE ON HISTORY_VACATION.employee_id=EMPLOYEE.ID
+                                    JOIN DEPARTMENT_EMPLOYEE ON EMPLOYEE.ID=DEPARTMENT_EMPLOYEE.employee_id
+                                    WHERE DEPARTMENT_EMPLOYEE.department_id = {self.department_id}
+                                    AND HISTORY_VACATION.finish_date > DATE('now')'''
+        if self.database.execute_sql(employees_on_vacation).fetchone()[0] >= 5:
             return print('Currently on vacation are 5 employees')
```

Comment:
There's common approach not to print the result, but return it.
For this methods raising exception would be better.
Create the your own class to manage this situation (5 employees on vacation), raise this exception instead of return smth and catch it in expected place.